### PR TITLE
BUG: fix isin with Series of tuples values (#16394)

### LINKED ISSUE #16394: BUG: ValueError with Series.isin and tuples
#### Code Sample, a copy-pastable example if possible

```
import pandas as pd

df = pd.DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'f']})
df['C'] = list(zip(df['A'], df['B']))
df['C'].isin([(1, 'a')])
```
#### Problem description
Returns ValueError:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda/envs/pandas_dev/lib/python3.6/site-packages/pandas/core/series.py", line 2555, in isin
    result = algorithms.isin(_values_from_object(self), values)
  File "/anaconda/envs/pandas_dev/lib/python3.6/site-packages/pandas/core/algorithms.py", line 421, in isin
    return f(comps, values)
  File "/anaconda/envs/pandas_dev/lib/python3.6/site-packages/pandas/core/algorithms.py", line 399, in <lambda>
    f = lambda x, y: htable.ismember_object(x, values)
  File "pandas/_libs/hashtable_func_helper.pxi", line 428, in pandas._libs.hashtable.ismember_object (pandas/_libs/hashtable.c:29677)
ValueError: Buffer has wrong number of dimensions (expected 1, got 2)

#### Expected Output
In pandas 0.19.2 returns:
0     True
1    False
2    False
Name: C, dtype: bool
#### Output of ``pd.show_versions()``

<details>
# Paste the output here pd.show_versions() here
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Darwin
OS-release: 16.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.0rc2
pytest: None
pip: 9.0.1
setuptools: 27.2.0
Cython: None
numpy: 1.12.1
scipy: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by jaredsnyder**
Switched out 
values = np.array(list(values), dtype='object')
for
values = lib.list_to_object_array(list(values))
 in the isin() method found in core/algorithms.py

Added test for comparing to a list of tuples

 - [ ] closes #16394
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -1201,6 +1201,13 @@ def test_isin_df(self):
         expected['B'] = False
         tm.assert_frame_equal(result, expected)
 
+    def test_isin_tuples(self):
+        df = pd.DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'f']})
+        df['C'] = list(zip(df['A'], df['B']))
+        result = df['C'].isin([(1, 'a')])
+        tm.assert_series_equal(result,
+                              Series([True,False,False],name="C"))
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_analytics.py]: for PEP8: the indentation does not fully match with the line before (one space more needed), and a space after the comma is needed

File: pandas/tests/frame/test_analytics.py [hunk 2]
```diff
@@ -1201,6 +1201,13 @@ def test_isin_df(self):
         expected['B'] = False
         tm.assert_frame_equal(result, expected)
 
+    def test_isin_tuples(self):
+        df = pd.DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'f']})
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_analytics.py]: Can you add the original issue number as a comment on the first line ? 
just like `# GH16394`

**Review by jorisvandenbossche — COMMENTED** [review]
The test looks good, let's wait what Travis says. 

Can you add a small entry about this fix in the whatsnew file? This can be in `doc/source/whatsnew/v0.20.2.txt`

File: pandas/tests/frame/test_analytics.py (hunk 2 above)
Comment (jaredsnyder) [inline pandas/tests/frame/test_analytics.py]: Fixed

File: pandas/tests/frame/test_analytics.py (hunk 1 above)
Comment (jaredsnyder) [inline pandas/tests/frame/test_analytics.py]: Sorry about that, fixed

**Comment by jaredsnyder** [comment]
Added the following to the whatsnew file under Bugs - Resizing:

- Bug in ``isin`` in core/algorithms.py with comparing lists of tuples

File: doc/source/whatsnew/v0.20.2.txt [hunk 3]
```diff
@@ -80,7 +80,7 @@ Reshaping
 ^^^^^^^^^
 
 - Bug in ``DataFrame.stack`` with unsorted levels in MultiIndex columns (:issue:`16323`)
-
+- Bug in ``isin`` in core/algorithms.py with comparing lists of tuples
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.2.txt]: ``Series.isin(..)`` with a list of tuples. add the issue number here as well.

**Review by jreback — APPROVED** [review]
lgtm, minor doc comment.

**Comment by jreback** [comment]
@jaredsnyder I think you just a couple of lint issues. https://travis-ci.org/pandas-dev/pandas/jobs/234970029

File: doc/source/whatsnew/v0.20.2.txt (hunk 3 above)
Comment (jaredsnyder) [inline doc/source/whatsnew/v0.20.2.txt]: Thanks, think I got the format correct with the latest commit

File: doc/source/whatsnew/v0.20.2.txt [hunk 4]
```diff
@@ -80,7 +80,7 @@ Reshaping
 ^^^^^^^^^
 
 - Bug in ``DataFrame.stack`` with unsorted levels in MultiIndex columns (:issue:`16323`)
-
+- Bug in ``Series.isin(..)`` in core/algorithms.py with a list of tuples (:issue:`16394`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.2.txt]: Bug in ``Series.isin(...)`` with a list of tuples. No need to mention internal things.

**Comment by jreback** [comment]
minor comment. ping on green.

**Comment by jaredsnyder** [comment]
@jreback addressed your comment

**Comment by jorisvandenbossche** [comment]
@jaredsnyder Thanks!

**Pull request merged**